diff --git a/node_modules/expo-modules-core/android/CMakeLists.txt b/node_modules/expo-modules-core/android/CMakeLists.txt
index 0000000..0000000 100644
--- a/node_modules/expo-modules-core/android/CMakeLists.txt
+++ b/node_modules/expo-modules-core/android/CMakeLists.txt
@@ -50,5 +50,5 @@ if(IS_NEW_ARCHITECTURE_ENABLED)
   add_subdirectory("${SRC_DIR}/fabric")
   set(NEW_ARCHITECTURE_DEPENDENCIES "fabric")
-  set(NEW_ARCHITECTURE_COMPILE_OPTIONS -DIS_NEW_ARCHITECTURE_ENABLED=1 -DRN_FABRIC_ENABLED=1 -DRN_SERIALIZABLE_STATE=1)
+  set(NEW_ARCHITECTURE_COMPILE_OPTIONS -DIS_NEW_ARCHITECTURE_ENABLED=1 -DRN_FABRIC_ENABLED=1)
 else()
   set(NEW_ARCHITECTURE_DEPENDENCIES "")
