diff --git a/node_modules/react-native-screens/android/CMakeLists.txt b/node_modules/react-native-screens/android/CMakeLists.txt
index 5d5dfe4..9fdefc1 100644
--- a/node_modules/react-native-screens/android/CMakeLists.txt
+++ b/node_modules/react-native-screens/android/CMakeLists.txt
@@ -34,6 +34,14 @@ target_compile_definitions(
     rnscreens
     PRIVATE
     -DFOLLY_NO_CONFIG=1
+    -D_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION
+)
+
+# Force undefine RN_SERIALIZABLE_STATE to avoid folly::dynamic errors
+target_compile_options(
+    rnscreens
+    PRIVATE
+    "-URN_SERIALIZABLE_STATE"
 )
 
 find_package(ReactAndroid REQUIRED CONFIG)
